src/js/helpers/ajaxWrapper.js

```diff
@@ -31,16 +31,23 @@ var qajaxWrapper = function (opts = {}) {
     body: null
   };
 
-  var parseResponse = function (xhr) {
+  var parseResponse = function (xhr, callback) {
     response.status = xhr.status;
-    try {
-      response.body = JSON.parse(xhr.responseText);
-    } catch (e) {
-      response.body = xhr.responseText;
-    }
+    xhr.json().then(function (body) {
+      response.body = body;
+    }, function () {
+      response.body = xhr.text().then(function (body) {
+        response.body = body
+      }, function () {
+        // couldn't read text from XHR
+        response.body = xhr.statusText;
```

Comment:
Yeah we could just leave this whole callback out. It was an attempt at doing something analogous to assigning the whole `responseText` to body (likely to be equally useless).